src/components/BookList.js

```diff
@@ -0,0 +1,48 @@
+import { v4 as uuidv4 } from 'uuid';
+import { useSelector, useDispatch } from 'react-redux';
+// import React from 'react';
+import { addBook } from '../redux/books/books';
+import Book from './Book';
+
+function BooksList() {
+  const booksStore = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+
+  const submitBookToStore = (e) => {
+    e.preventDefault();
+
+    const id = uuidv4();
+    const titleInput = document.getElementById('books-input');
+    const authorInput = document.getElementById('author-input');
+
+    const newBook = {
+      id,
+      title: titleInput.value,
+      author: authorInput.value,
+    };
+
+    dispatch(addBook(newBook));
+
+    titleInput.value = '';
+    authorInput.value = '';
+  };
+
+  return (
+    <div className="container">
+      <ul>
+        {
+          booksStore.map((book) => (
+            <Book key={book.id} book={book} />
+          ))
+        }
+      </ul>
+      <form action="" onSubmit={submitBookToStore}>
+        <input type="text" name="" placeholder="Book" id="books-input" required />
+        <input type="text" name="" placeholder="Author" id="author-input" required />
+        <button type="submit">Add Book</button>
+      </form>
```

Comment:
- [ ] In keeping with the SOLID principle, kindly extract this new book form and its other functionalities into a separate component.
- [ ] Your inputs are not controlled. Your inputs should be controlled and states should be the only source of truth. You can refer to the first paragraph of this [article](https://reactjs.org/docs/forms.html) to get an idea of what this means. As listed in the project requirements (see image below) your add book form component should have a local `state` for `title` and `author`. And that local state should be updated whenever a user updates the title and author inputs with `onchange`. You can refer to this [article](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/) for a quick solution on how to create controlled components with React hooks.

  ![muth-req](https://user-images.githubusercontent.com/36542986/146150850-cbd4e1b0-84a1-448b-8227-102f4459f7fe.jpg)